physcraper/__init__.py

```diff
@@ -2261,85 +2323,100 @@ def remove_identical_seqs(self):
-                            debug("wrong tax_id given by ncbi?")
-                            ncbi_id = self.ids.ncbi_parser.get_id_from_name(tax_name)
-                            self.data.gb_dict[gb_id]['staxids'] = ncbi_id
-                            input_rank_id = self.ids.ncbi_parser.match_id_to_mrca(ncbi_id, list(self.ncbi_mrca)[0])#TODO HACK! will break mrca list
-
-                        # #######################################################
-                        if int(input_rank_id) == int(list(self.ncbi_mrca)[0]):  # belongs to ingroup mrca -> add to data, if not, leave it out
-                            # debug("input belongs to same mrca")
+            added = False
+            assert gb_id in self.data.gb_dict.keys(), (gb_id, self.data.gb_dict.keys())
+            if gb_id not in all_added_gi:
+                reason = ""
+                all_added_gi.add(gb_id)
+                # if len(gb_id.split(".")) == 1:
+                #     debug(gb_id)
+                if self.blacklist is not None and gb_id in self.blacklist:
+                    debug("gb_id in blacklist, not added")
+                    pass
+                elif gb_id in self.newseqs_acc:  # added to increase speed. often seq was found in another blast file
+                    debug("passed, was already added")
+                    pass
+                else:
+                    # debug(len(seq.replace("-", "").replace("N", "")))
+                    # debug(seq_len_cutoff)
+                    if len(seq.replace("-", "").replace("N", "")) > seq_len_cutoff:
+                        # debug(self.config.blast_loc)
+                        if self.config.blast_loc != "remote":
+                            # debug("ADD")
+                            tax_name = None
+                            
+                            # rank_mrca_ncbi = self.ids.ncbi_parser.get_rank(mrca_ncbi)
+                            # get rank to delimit seq to ingroup_mrca
+                            ncbi_id, tax_name = get_taxinfo_from_acc(gb_id, self.data, self.ids)
+                            rank_mrca_ncbi = set()
+                            # print(self.mrca_ncbi_list)
+                            # print(self.ids.mrca_ncbi)
+                            for mrca in self.mrca_ncbi_list:
+                                rank_mrca = self.ids.ncbi_parser.get_rank(mrca)
+                                rank_mrca_ncbi.add(rank_mrca)
+                                try:  # sometimes ncbi has wrong id linked
+                                    input_rank_id = self.ids.ncbi_parser.match_id_to_mrca(ncbi_id, mrca)
+                                except IndexError:  # this is for the wrong ncbi link, get right tax_id and do search again
+                                    debug("wrong tax_id given by ncbi? tax_id: {}, tax_name:{}".format(ncbi_id, tax_name))
+                                    ncbi_id = self.ids.ncbi_parser.get_id_from_name(tax_name)
```

Comment:
Here we already have the ncbi_id for the sequence from line 2368, so we don't need to look it up again.

Reply:
(I know this was already in my version -just checking that it can be deleted.)

Reply:
Well, sometimes the local blast gives a taxon id which is different from the one it should be, that's why we redefine it here.

Reply:
Ah, how do we know it is the wrong taxon id? It has an id that isn't in the ncbi taxonomy anywhere? I think in that case the safest thing may be to just drop the sequence... but maybe there is something I am missing?